pkg/cni/cilium/cilium.go

```diff
@@ -353,13 +355,18 @@ func ValidateValuesUpdate(newValues, oldValues map[string]any, fieldPath *field.
 	}
 	allErrs = append(allErrs, validateImmutableValues(newOperator, oldOperator, operPath, []string{
 		"securityContext",
-	})...)
+	}, []string{})...)
 
 	return allErrs
 }
 
-func validateImmutableValues(newValues, oldValues map[string]any, fieldPath *field.Path, immutableValues []string) field.ErrorList {
+func validateImmutableValues(newValues, oldValues map[string]any, fieldPath *field.Path, immutableValues []string, excludedKeys []string) field.ErrorList {
 	allErrs := field.ErrorList{}
+	for _, exclusion := range excludedKeys {
+		if excludedKeyExists(newValues, strings.Split(exclusion, ".")...) {
+			return allErrs
+		}
+	}
```

Comment:
Doesn't this completely skip the validation for the whole list of keys if any sub key is excluded? e.g. this is called with `immutableValues := []string{"cni", "ipam", "ipv6"}`, but since `excludedKeys := []string{"cni.chainingMode", "ipam.operator.clusterPoolIPv4PodCIDR"} is set, it will find `ipam.operator.clusterPoolIPv4PodCIDR` in `newValues` and just return this function without validating _any_ keys configured as immutable?

Reply:
That's a good concern, if we want to go that way, I also thought that we could validate each of them separately.